target.mk: further improve handling of default enabled SECCOMP

sch-m opened the pull request:
@Ansuel 
The fix in commit 847fad476f3d ("target.mk: improve handling of default enabled SECCOMP") unfortunately does not work for targets where the ARCH variable is set in ./$(SUBTARGET)/target.mk.

To get this working, the ./$(SUBTARGET)/target.mk must be included before the check.

In addition, the fix should not only take effect if no .config file exists, but also if a .config exists but the CONFIG_SECCOMP option is missing in the file.

This is relevant, for example, if you are working with .config templates and then want to complete the configuration using make defconfig.

sch-m committed:
target.mk: further improve handling of default enabled SECCOMP

The fix in commit 847fad476f3d ("target.mk: improve handling of default
enabled SECCOMP") unfortunately does not work for targets where the ARCH
variable is set in ./$(SUBTARGET)/target.mk.

To get this working, the ./$(SUBTARGET)/target.mk must be included
before the check.

Fixes: 847fad476f3d ("target.mk: improve handling of default enabled SECCOMP")
Signed-off-by: Martin Schiller <ms@dev.tdt.de>

sch-m commented:
> By grepping for the CONFIG you will always enable SECCOMP even if you want that disabled. 

Why do you think so? If SECCOMP is disabled, there is a `# CONFIG_SECCOMP is not set` in the .config and nothing will be changed.

In my tests this works as expected.

Ansuel commented:
```
($(shell grep "CONFIG_SECCOMP" $(TOPDIR)/.config 2>/dev/null),)
```
will be true for both the config enabled or not

sch-m commented:
> @sch-m ok now it's clear.
> 
> Can you by chance split this in 2 commit, one fixing the target.mk and the other reworking the logic? Also would be good to have some extra comments to descript this change.

OK, I'll split this into 2 commits.
> 
> Also did you test that grep doesn't print error in the context of a .config not existing?

The `2>/dev/null` will suppress any errors.

sch-m committed:
target.mk: init default configs if they are missing in the .config

The config options that are enabled by default and where other default
packages depends on should not only be set if there is no .config file,
but also if the .config exists but the config option (e.g.
CONFIG_SECCOMP) is missing in the file.

This is relevant, for example, if you are working with .config templates
and then want to complete the configuration using make defconfig.

Signed-off-by: Martin Schiller <ms@dev.tdt.de>

Ansuel commented:
@sch-m can you rebase on top of main or permit maintainers to push to pr?

sch-m commented:
> @sch-m can you rebase on top of main or permit maintainers to push to pr?

Done.
